### ISSUE TITLE
This plugin is not configuration cache compatible starting with Gradle 7.5

### ISSUE THREAD

**Issue by liutikas** [gradle/common-custom-user-data-gradle-plugin#44]
This plugin is not configuration cache compatible starting with Gradle 7.5.

When we upgraded androidx project to 7.5 nightly, our builds started failing due to this plugin calling external processes and that breaking configuration

https://ge.androidx.dev/s/zd2jedmewi6pm/console-log#L566

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git --version' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndCheckSuccess(Utils.java:153)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.isGitInstalled(CustomBuildScanEnhancements.java:330)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:282)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git config --get remote.origin.url' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndGetStdOut(Utils.java:169)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:286)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git rev-parse --abbrev-ref HEAD' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndGetStdOut(Utils.java:169)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.lambda$execute$0(CustomBuildScanEnhancements.java:289)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.getGitBranchName(CustomBuildScanEnhancements.java:340)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:289)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git rev-parse --short=8 --verify HEAD' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndGetStdOut(Utils.java:169)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:288)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git rev-parse --verify HEAD' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndGetStdOut(Utils.java:169)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:287)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

```
org.gradle.api.InvalidUserCodeException: Starting an external process 'git status --porcelain' during configuration time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onExternalProcessStarted(ConfigurationCacheProblemsListener.kt:78)
	at org.gradle.configurationcache.InstrumentedInputAccessListener.externalProcessStarted(InstrumentedInputAccessListener.kt:110)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:389)
	at org.gradle.internal.classpath.Instrumented.externalProcessStarted(Instrumented.java:393)
	at org.gradle.internal.classpath.Instrumented.exec(Instrumented.java:258)
	at com.gradle.Utils.execAndGetStdOut(Utils.java:169)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:290)
	at com.gradle.CustomBuildScanEnhancements$CaptureGitMetadataAction.execute(CustomBuildScanEnhancements.java:270)
	at com.gradle.enterprise.gradleplugin.internal.extension.b$4.run(SourceFile:185)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

**Comment by etiennestuder** [comment]
We are aware of this issue and expect a fix on the build tool side.

**Comment by runningcode** [comment]
This is fixed on the build tool side for Gradle 7.5.
